pkg/filter/example_select_test.go

```diff
@@ -0,0 +1,137 @@
+package filter_test
+
+import (
+	"fmt"
+	"log"
+
+	bundle "github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/apis/bundle/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/converter"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/filter"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var componentsYAMLs = `
+components:
+- spec:
+    componentName: zap
+    objects:
+    - apiVersion: v1
+      kind: Pod
+      metadata:
+        name: zap-pod
+        labels:
+          component: zork
+        annotations:
+          foo: bar
+        namespace: kube-system
+    - apiVersion: v1
+      kind: Pod
+      metadata:
+        name: zip-pod
+        labels:
+          component: zork
+        annotations:
+          foo: baz
+        namespace: default
+- spec:
+    componentName: bog
+    objects:
+    - apiVersion: v1
+      kind: Pod
+      metadata:
+        name: bog-pod-1
+        labels:
+          component: bork
+        annotations:
+          foof: yar
+        namespace: kube-system
+    - apiVersion: v1
+      kind: Pod
+      metadata: 
+        name: bog-pod-2
+        namespace: default
+- spec:
+    componentName: nog
+    objects:
+    - apiVersion: v1beta1
+      kind: Pod
+      metadata:
+        name: nog-pod
+        labels:
+          component: nork
+        annotations:
+          foof: narf
+        namespace: kube
+- spec:
+    componentName: zog
+    objects:
+    - apiVersion: v1
+      kind: Pod
+      metadata:
+        name: zog-pod
+        namespace: kube-system
+    - apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: zog-dep
+        labels:
+          component: zork
+        annotations:
+          zoof: zarf
+        namespace: zube
+- spec:
+    componentName: kog
+    objects:
+    - apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: kog-dep
+        namespace: kube-system`
+
+func componentName(c *bundle.Component) string {
+	return c.Spec.ComponentName
+}
+
+type component struct {
+	name    string
+	objects []string
+}
+
+func toComponent(comp *bundle.Component) component {
+	c := component{name: componentName(comp)}
+	for _, obj := range comp.Spec.Objects {
+		c.objects = append(c.objects, obj.GetName())
+	}
+	return c
+}
+
+func toComponentList(components []*bundle.Component) []component {
+	out := make([]component, len(components))
+	for i, c := range components {
+		out[i] = toComponent(c)
+	}
+	return out
+}
+
+func Example_select() {
```

Comment:
This isn't a test method...?

Also, I would recommend a table test that has more extensive cases.

Reply:
Oh, I see, it's an example. I've honestly never used examples.

Still, some table tests might be nice.